tools/maven/checkstyle.xml

```diff
@@ -0,0 +1,317 @@
+        <property name="format" value="\s+$"/>
+        <property name="message" value="Trailing whitespace"/>
+        <property name="severity" value="error"/>
+    </module>
+
+    <module name="RegexpSingleline">
+        <property name="format" value="Throwables.propagate\("/>
+        <property name="message" value="Throwables.propagate is deprecated"/>
+        <property name="severity" value="error"/>
+    </module>
+
+    <module name="SuppressionFilter">
+        <property name="file" value="${checkstyle.suppressions.file}" default="suppressions.xml"/>
+    </module>
+
+    <!--
+
+    Amoro CUSTOM CHECKS
+
+    -->
+
+    <module name="FileLength">
+        <property name="max" value="3000"/>
+    </module>
+
+    <!-- All Java AST specific tests live under TreeWalker module. -->
+    <module name="TreeWalker">
+
+        <!-- Allow use of comment to suppress javadocstyle -->
+        <module name="SuppressionCommentFilter">
+            <property name="offCommentFormat" value="CHECKSTYLE.OFF\: ([\w\|]+)"/>
+            <property name="onCommentFormat" value="CHECKSTYLE.ON\: ([\w\|]+)"/>
+            <property name="checkFormat" value="$1"/>
+        </module>
+
+        <!--
+
+        Amoro CUSTOM CHECKS
+
+        -->
+
+        <!-- Prohibit T.getT() methods for standard boxed types -->
+        <module name="Regexp">
+            <property name="format" value="Boolean\.getBoolean"/>
+            <property name="illegalPattern" value="true"/>
+            <property name="message" value="Use System.getProperties() to get system properties."/>
+        </module>
+
+        <module name="Regexp">
+            <property name="format" value="Integer\.getInteger"/>
+            <property name="illegalPattern" value="true"/>
+            <property name="message" value="Use System.getProperties() to get system properties."/>
+        </module>
+
+        <module name="Regexp">
+            <property name="format" value="Long\.getLong"/>
+            <property name="illegalPattern" value="true"/>
+            <property name="message" value="Use System.getProperties() to get system properties."/>
+        </module>
+
+        <!--
+
+        IllegalImport cannot blacklist classes so we have to fall back to Regexp.
+
+        -->
+
+        <!-- forbid the use of google.common.base.Preconditions -->
+        <module name="Regexp">
+            <property name="format" value="import com\.google\.common\.base\.Preconditions"/>
+            <property name="illegalPattern" value="true"/>
+            <property name="message"
+                      value="Do not use Guava's Preconditions"/>
+        </module>
+
+        <module name="Regexp">
+            <property name="format" value="import static com\.google\.common\.base\.Preconditions"/>
+            <property name="illegalPattern" value="true"/>
+            <property name="message"
+                      value="Do not use Guava's Preconditions directly"/>
```

Comment:
Can we add some guidance in the message to instruct developers on how to solve this style error?